### ISSUE TITLE
ensure basic attestation type allows for self-signed attestation certs

### ISSUE THREAD

**Issue by equalsJeffH** [w3c/webauthn#1498]
some authenticators in the wild return, on registration, an attestation certificate that is signed by the private key associated with the public key in the cert. Thus, the attestation certificate is _essentially_ the "(CA) root certificate".  Verifying such a certificate can be accomplished by, e.g.: 
```
$ cd /some-directory    # where the cert files reside...
$ openssl verify -CAfile ./self-signed-cert.pem -CApath /some-directory ./self-signed-cert.pem
./self-signed-cert.pem: OK
```
...though the value of verifying the signature is debatable.  If the RP has obtained the putative "root certificate" via out-of-band means (e.g., via the FIDO Alliance Metadata Service), they can also "verfiy" the returned attestation cert against the putative "root certificate" by a simple byte-by-byte comparison (once the certificates are both represented in the same format, e.g., DER or PEM).

This key pair is used only for attestation (unlike the present "self-attestation" attstn type, where the user's per-RP key pair is used to construct a one-off per-RP attestation cert) and thus the same attestation cert is returned to all RPs. Thus we ought to define a new attestation type for this.

Since this is fairly similar to Basic Attestation, I suggest:  "Self-signed Basic Attestation"

**Comment by sbweeden** [comment]
I think a more explicit description of why it is believed this needs to be differentiated from "Basic" attestation might help with a more open discussion. My limited understanding is that this somehow ties back in to RFC5280, but I am not sure where (or if) WebAuthn defines that "Basic" attestation x5c verification must always follow RFC5280. Is this explicit anywhere?

Step 21 (3rd sub-bullet) of section 7.1 of the current editors draft uses the text: 
"Otherwise, use the X.509 certificates returned as the attestation trust path from the verification procedure to verify that the attestation public key correctly chains up to an acceptable root certificate". 

RFC5280 might surely be one (even the preferred) way to do that, but couldn't we simply indicate that for self-signed attestation certs there are other ways this might be done without having to introduce a new type of attestation definition?

**Comment by equalsJeffH** [comment]
> I think a more explicit description of why it is believed this needs to be differentiated from "Basic" attestation might help with a more open discussion. 

the WG has decided (on the 2020-10-21 call) that a "self-signed" attestation cert does not require an attestation type other than basic. I've closed PR #1499. will open a new PR that simply clarifies the self-signed case.

**Comment by equalsJeffH** [comment]
see new PR #1509
